src/target/linux.rs

```diff
@@ -404,6 +431,65 @@ impl LinuxTarget {
         Err(Box::new(HardwareBreakpointError::UnsupportedPlatform))
     }
 
+    /// Set a breakpoint at a given address
+    /// This modifies the Target's memory by writting an INT3 instr. at `addr`
+    /// The returned Breakpoint object can then be used to disable the breakpoint
+    /// or query its state
+    pub fn set_breakpoint(&self, addr: usize) -> Result<Breakpoint, Box<dyn std::error::Error>> {
+        let bp = Breakpoint::new(addr, self.pid())?;
+        let existing_breakpoint = {
+            let hdl = self.breakpoints.borrow();
+            hdl.get(&addr).and_then(|val| Some(val.clone()))
+        };
+
+        let mut bp = match existing_breakpoint {
+            None => {
+                self.breakpoints.borrow_mut().insert(addr, bp);
+                bp
+            }
+            // If there is already a breakpoint set, we give back the existing one
+            Some(breakpoint) => breakpoint.clone(),
+        };
+        bp.set()?;
+        //self.breakpoints.borrow_mut().insert(addr, bp);
+        Ok(bp)
+    }
+
+    pub fn single_step(&self) -> Result<nix::sys::wait::WaitStatus, Box<dyn std::error::Error>> {
```

Comment:
`UnixTarget` already provides this and `unpause`. Could you move the breakpoint handling there or if easier remove them from `UnixTarget` for now?